vulcanai2/models/basenetwork.py

```diff
@@ -99,50 +103,50 @@ def __init__(self, name, dimensions, config, save_path=None, input_networks=None
-                for net in self._input_networks:
-                    self.in_dim.append(net.in_dim)
-        #import pdb; pdb.set_trace()
-
         self._create_network(
             activation=activation,
             pred_activation=pred_activation)
-       
-        out_shapes = self.get_output_shapes()
-        self.out_dim = out_shapes[list(out_shapes)[-1]]['output_shape'][1:] 
-
-    # TODO: ignore for now
-    # def get_in_dim(self, network = None):
-    #     if not network:
-    #         network = self
-    #     for net in network.input_networks:
-    #             if net.input_networks is None:
-                    
-    #                 self.in_dim.append(net.get_output_shapes())
-
-    #             else:
-    #                 self.in_dim = self.get_in_dim(net)
-    #             temp_tensor = torch.zeros([1, *net.in_dim])
+
+        #print(self)
+        
+        out_shapes = self.get_output_shapes(network=self.network,input_size=self.in_dim)
+        self.out_dim = out_shapes[list(out_shapes)[-1]]['output_shape'][1:]
+        if len(self.out_dim)>1:
+            self.out_dim = tuple(self.out_dim)
+        else:
+            self.out_dim = self.out_dim[0]
+        if self._num_classes:
+            out_shapes = self.get_output_shapes(network=self.network_tail,input_size=self.out_dim)
+            self.out_dim = out_shapes[list(out_shapes)[-1]]['output_shape'][1:]
+        
+        #print(self.out_dim)
+
     # TODO: where to do typechecking... just let everything fail?
 
     def forward(self, inputs):
-        """Perform a forward pass through the module/modules."""
-        if self._input_networks:
-            if not isinstance(inputs, list):
+        """
+        Perform a forward pass through the module/modules.
+        If the network is defined with `num_classes` then it is
+        assumed to be the last network which contains a classification
+        layer/network tail. The inputs will be passed
+        through the networks and then through the classifier.
+        If not, the input is passed through the network and
+        returned without passing through a classification layer.
+        :param x: input torch.Tensor
+        :return: output torch.Tensor        
+        """
+        if not isinstance(inputs, list):
                 inputs = [inputs]
+        bs = inputs[0][0]
+        if self._input_networks is not None:
             models=self._input_networks
-            
             outputs = []
             for model, x in zip(models, inputs):
                 outputs.append(model(x))
-            network_output = self._forward(torch.cat(outputs))
+                #TODO: Use tablemodules NEW: https://github.com/torch/nn/blob/master/doc/table.md
+            network_output = self._forward(torch.s(outputs, dim=1)) # Ref:https://github.com/torch/torch7/blob/master/doc/maths.md
         else:
-            network_output = self._forward((inputs))
+            network_output = self._forward((inputs[0]))
```

Comment:
you don't need to do this if only one model is passed. just treat it as if there are many passed